sourcegraph/sourcegraph
# web: add time-since-navigation and time-since-app-mount to spans

## Context

This PR adds `time.since_navigation` and `time.since_app_mount` to every client span. We can leverage these attributes to create valuable dashboards in Honeycomb later.

Closes https://github.com/sourcegraph/sourcegraph/issues/44008

## Test plan

1. `ENABLE_OPEN_TELEMETRY=true sg start web-standalone`
2. Open the browser console and ensure that debug-level output is enabled.
3. Reload any web application page.
4. See new attributes added to spans.

<img width="588" alt="Screenshot 2022-11-07 at 17 06 32" src="https://user-images.githubusercontent.com/3846380/200270828-2aefadfa-8511-4780-83d3-eddb77347470.png">

## App preview:

- [Web](https://sg-web-vb-time-since-shared-span-start.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
